libtransistor.mk

```diff
@@ -4,13 +4,20 @@ else
     LIBTRANSISTOR_HOME := $(realpath $(LIBTRANSISTOR_HOME))
 endif
 
+SYS_INCLUDES := -isystem $(LIBTRANSISTOR_HOME)/newlib/newlib/libc/include/ -isystem $(LIBTRANSISTOR_HOME)/newlib/newlib/libc/sys/switch/include/
+INCLUDES := $(SYS_INCLUDES) -I$(LIBTRANSISTOR_HOME)/include/ -I$(LIBTRANSISTOR_HOME)/libssp/include
+WARNINGS := -Wall -Wextra
+
 LD := ld.lld$(LLVM_POSTFIX)
 CC := clang$(LLVM_POSTFIX)
 CXX := clang++$(LLVM_POSTFIX)
 AS := llvm-mc$(LLVM_POSTFIX)
+AR := llvm-ar$(LLVM_POSTFIX)
 LD_FLAGS := -Bsymbolic --shared --emit-relocs --no-gc-sections --no-undefined -T $(LIBTRANSISTOR_HOME)/link.T
-CC_FLAGS := -g -fPIC -ffreestanding -fexceptions -target aarch64-none-linux-gnu -O0 -mtune=cortex-a53 -I "$(LIBTRANSISTOR_HOME)/include/" -isystem "$(LIBTRANSISTOR_HOME)/newlib/newlib/libc/include/" -isystem "$(LIBTRANSISTOR_HOME)/newlib/newlib/libc/sys/switch/include/" -I$(LIBTRANSISTOR_HOME)/libssp/include -Weverything -Wno-missing-prototypes -Wno-strict-prototypes -Wno-sign-conversion -Wno-missing-variable-declarations -Wno-unused-parameter -Wno-cast-align -Wno-padded -Wno-cast-qual -Wno-gnu-binary-literal -Werror-implicit-function-declaration -Wno-format-pedantic
-CFLAGS := $(CC_FLAGS) # for compatiblity
+CC_FLAGS := -g -fPIC -ffreestanding -fexceptions -O0 -mtune=cortex-a53 -target aarch64-none-linux-gnu -nostdlib -nostdinc $(INCLUDES) $(WARNINGS)
+AR_FLAGS := rcs
+# for compatiblity
```

Comment:
Definitely should've had this comment on its own line. Make must've outsmarted us and stripped this comment.